nc/creds_fetcher.go

```diff
@@ -0,0 +1,73 @@
+package nc
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/config"
+	"github.com/NordSecurity/nordvpn-linux/core"
+)
+
+type Time interface {
+	GetSecondsSinceTimestamp(int64) float64
+}
+
+type RealTime struct{}
+
+func (r RealTime) GetSecondsSinceTimestamp(timestamp int64) float64 {
+	return time.Since(time.Unix(timestamp, 0)).Seconds()
+}
+
+type CredentialsGetter struct {
+	api  core.CredentialsAPI
+	cm   config.Manager
+	time Time
```

Comment:
Is this `lastFetchTime` ? if keeping in stile with short names, then comment about field meaning would be nice

Reply:
or is this `validityPeriod` ?

Reply:
This is an interface for getting the time, introduced so that it can be mocked in uts. I have changed the name so that it is more meaningful.